CDCgov/prime-reportstream
# Experience/6341: return Valid status if warnings but no errors are submitted

This PR returns Valid status (http 201) if warnings but no errors are submitted to the `/api/validate` endpoint. This makes the endpoint's behavior more consistent with the actual file submission endpoint  (`/api/waters`).

Test Steps:
1. Log into RS as prime admin
2. Make sure the "validation-service" feature flag is set
3. Set your org and choose "Validate" from the "Upload & Validate" menu
4. Open up your browser's web tools and go to the Network tab
5. Upload your sample HL7 or CSV file that only throws warnings (not errors)
6. Observe that the "validate" network call now returns a 201 in the Network tab, indicating a Valid status

### Testing

## Linked Issues
- Fixes #6341 
- Relates to #5713 and #6285
